syntax = "proto3";

option java_multiple_files = true;
option java_outer_classname = "PosixIo";
option java_package = "de.zib.paciofs.io.posix.grpc";

package paciofs.io.posix.grpc;

import "messages/ping.proto";
import "messages/stat.proto";

message PingRequest {
  messages.Ping ping = 1;
}

message PingResponse {
  messages.Ping ping = 1;
}

message StatRequest {
  string path = 1;
}

message StatResponse {
  messages.Stat stat = 1;
}

service PosixIoService {
  rpc Ping(PingRequest) returns (PingResponse);
  rpc Stat(StatRequest) returns (StatResponse);
}
